pkg/build/controller/controller.go

```diff
@@ -289,6 +291,81 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			return fmt.Errorf("failed to update build %s/%s: %v", build.Namespace, build.Name, err)
 		}
 		glog.V(4).Infof("Build %s/%s status was updated %s -> %s", build.Namespace, build.Name, build.Status.Phase, nextStatus)
+
+		if buildutil.IsBuildComplete(build) {
+			if err := bc.executeBuildPostHooks(build); err != nil {
+				// Do not return the error, since errors executing the
+				// post hooks should not influence the build lifecycle.
+				util.HandleError(fmt.Errorf("failed to execute post hooks of build %s/%s: %v", build.Namespace, build.Name, err))
+			} else {
+				glog.V(4).Infof("Post hooks of build %s/%s were executed successfully", build.Namespace, build.Name)
+			}
+		}
+	}
```

Comment:
> we need controller logic to also re-process builds to ensure the post-actions get invoked

We discussed in planning that there would be no retry logic. Once the build completes we try to execute the actions (at most) once. We can add retry in the future by introducing a new field to specific the retry policy, just like deployment hooks have. I have in mind leaving room for that.